intake/__init__.py

```diff
@@ -145,5 +154,10 @@ def open_catalog(uri=None, **kwargs):
     if "_file" not in driver:
         kwargs.pop("fs", None)
     if driver not in registry:
-        raise ValueError("Unknown catalog driver (%s), supply one of: %s" % (driver, list(sorted(registry))))
+        raise ValueError(
+            f"Unknown catalog driver '{driver}'. "
+            "Do you need to install a new driver from the plugin directory? "
+            "https://intake.readthedocs.io/en/latest/plugin-directory.html\n"
+            f"Current registry: {list(sorted(registry))}"
```

Comment:
We could in theory filter this list (which might be long) for only those that derive from `Catalog`. That would mean actually importing them, so I am slightly against it.

Reply:
That might be less confusing, I agree. But yeah we'd possibly need to import a bunch of stuff just to display an error. Alternatively we don't have to show the list at all and could say something like `"Use list(registry) to see currently loaded drivers"`.

Getting into over-engineering territory here, but the `registry` could also have dynamic a `.catalog_drivers` store.

Reply:
> Getting into over-engineering territory here, but the registry could also have dynamic a .catalog_drivers store.

Maybe if someone asks for it! Most catalog drivers have "cat" in the name anyway, so on reflection, I think we should leave it as it is.